### DISCUSSION TITLE
python computation without blocking js event loop

### DISCUSSION THREAD

**Discussion by beauxq** [pyscript/pyscript#1406] [Q&A]
I see this: https://github.com/pyscript/pyscript/pull/1333
I don't see documentation on how to use it.

Does this make it so I can't call python from the JS main thread?

Being able to do something like this from the JS main thread would be nice:
where `long` is a normal (not async) function defined in python
```js
const python_long_function = pyscript.interpreter.globals.get_async('long');

await python_long_function();
// runs sync python function `long` in worker thread
// and finishes this promise in main thread when it's complete
```

How can I do something like this?

**Comment by WebReflection** [comment]
I think @hoodmane would be the most indicate person to fully answer your question but there's current work on making blocking sync code working from Web Workers and your use case *should* be covered once we release this.

**Comment by hoodmane** [comment]
From an implementation perspective it is not hard to provide an interface as follows:
```js
const blah = await pyscript.interpreter.globals.get('blah');
await blah()
```
Maybe we will provide this. I am not sure what the plan is.
